src/Api/Controllers/UsersController.php

```diff
@@ -244,4 +246,40 @@ public function changeAppUserActiveStatus(int $id, int $appsId) : Response
         $userAssociatedToApp->updateOrFail();
         return $this->response($userAssociatedToApp);
     }
+
+    /**
+     * unsubscribe from notification
+     *
+     * @param int $id
+     * @throws InternalServerErrorException
+     *
+     * @return Response
+     */
+    public function unsubscribe(int $id) : Response
+    {
+        $request = $this->request->getPostData();
+
+        
+        if (!isset($request['notification_types'])) {
+            throw new Exception("Error Processing Request", 1);
+        }
+        
+        //none admin users can only edit themselves
+        if (!$this->userData->hasRole('Default.Admins') || $id == 0) {
+            $id = $this->userData->getId();
+        }
+
+        $user = $this->model->findFirstOrFail([
+            'id = ?0 AND is_deleted = 0',
+            'bind' => [$id],
+        ]);
+        $unsubscribe = [];
+        foreach ($request['notification_types'] as $typeId) {
+            $notificationType = NotificationType::findFirst($typeId);
+            $systemModulesId = $notificationType ? $notificationType->system_modules_id : -1;
+            $unsubscribe[] = Notifications::unsubscribe($user, (int) $typeId, (int) $systemModulesId);
```

Comment:
@dasdo its good for now ? but shouldnt this metho of unbuscribe be on the users end?

Im a user who is unsubscribing from the type X?

so it should be $user->unsubcribe(NotificationType)?